transformer_engine/pytorch/transformer.py

```diff
@@ -77,6 +77,48 @@ def forward(self, hidden_state: torch.Tensor) -> torch.Tensor:
         output = hidden_state.div(keep_prob) * random_tensor
         return output
 
+class _SplitLastDim(torch.autograd.Function):
+    """"""
+
+    @staticmethod
+    def forward(ctx,
+                mixed_x_layer: torch.Tensor,
+                num_parts: int
+    ) -> Tuple[torch.Tensor, ...]:
+        return split_tensor_along_dim(mixed_x_layer, -1, num_parts)
+
+    @staticmethod
+    def backward(ctx,
+                 *grad_outputs):
+        assert len(grad_outputs) > 0, "No gradients received for backprop!"
+
+        noop_ok = True
+        strides = grad_outputs[0].stride()
+        data_ptr = grad_outputs[0].untyped_storage().data_ptr()
+        shape = grad_outputs[0].shape
+        last_dim_size = grad_outputs[0].shape[-1]
+        for i, tensor in enumerate(grad_outputs):
+            if (tensor.stride() != strides or
+                tensor.shape != shape or
+                tensor.untyped_storage().data_ptr() != data_ptr or
+                tensor.storage_offset() != i * last_dim_size):
+                noop_ok = False
+                break
+
+        if noop_ok:
+            ret = torch.Tensor().to(grad_outputs[0].dtype)
+            if grad_outputs[0].is_cuda:
+                ret = ret.cuda()
```

Comment:
Slightly more robust:
```suggestion
            ret = torch.Tensor().to(device=args[0].device,
                                    dtype=args[0].dtype)
```